Tasks/PowerShellOnTargetMachines2/PowerShellOnTargetMachines.ps1

```diff
@@ -0,0 +1,43 @@
+Trace-VstsEnteringInvocation $MyInvocation
+$global:ErrorActionPreference = 'Continue'
+
+# Undocumented VstsTaskSdk variable so Verbose/Debug isn't converted to ##vso[task.debug].
+# Otherwise any content the ad-hoc script writes to the verbose pipeline gets dropped by
+# the agent when System.Debug is not set.
+$global:__vstsNoOverrideVerbose = $true
+
+try {
+    Import-VstsLocStrings "$PSScriptRoot\task.json"
+    
+    . "$PSScriptRoot\Utility.ps1"
+    . "$PSScriptRoot\SessionHelper.ps1"
+    . "$PSScriptRoot\JobHelper.ps1"
+    . "$PSScriptRoot\RunPowerShellScriptJob.ps1"
+
+    # Get all inputs for the task
+    $input_Machines = Get-VstsInput -Name "Machines" -Require -ErrorAction "Stop"
+    $input_UserName = Get-VstsInput -Name "UserName" -Require -ErrorAction "Stop"
+    $input_UserPassword = ConvertTo-SecureString -AsPlainText -String $(Get-VstsInput -Name "UserPassword" -Require -ErrorAction "Stop") -Force
+
+    $input_Protocol = Get-VstsInput -Name "Protocol" -Require -ErrorAction "Stop"
+    $input_NewPsSessionOptionArguments = Get-VstsInput -Name "NewPsSessionOptionArguments"
+
+    $input_RunPowershellInParallel = Get-VstsInput -Name "RunPowershellInParallel" -AsBool
+
+    $targetMachineNames = Parse-TargetMachineNames -machineNames $input_Machines
```

Comment:
Your task input recommends providing machinename:port. I see the machine name with port is passed along down for creating PS session.

Reply:
yes true, I need to make that change to handle the case when port is provided as part of machine name.